app/scripts/lib/app-start.js

```diff
@@ -706,7 +708,26 @@ function (
     },
 
     _getContext: function () {
-      return this._searchParam('context') || Constants.DIRECT_CONTEXT;
+      if (this._isVerification()) {
+        return this._getVerificationContext();
+      } else {
+        return this._searchParam('context') || Constants.DIRECT_CONTEXT;
+      }
+    },
+
+    _getVerificationContext: function () {
+      // Users that verify in the same browser use the same context that
+      // was used to sign up to allow the verification tab to have
+      // the same capabilities as the signup tab.
+      // If verifying in a separate browser, fall back to the default context.
+      var verificationInfo = this._getVerficationInfo();
+      return verificationInfo.get('context') || Constants.DIRECT_CONTEXT;
+    },
+
+    _getVerficationInfo: function () {
```

Comment:
Add a test for this function, and rename to `_getSameBrowserVerificationInfo` or similar.